src/mesh/amr_loadbalance.cpp

```diff
@@ -1194,22 +1154,310 @@ void Mesh::FinishRecvCoarseToFineAMR(MeshBlock *pb, Real *recvbuf) {
+    int xf3 = static_cast<int>(nloc.lx3 & 1LL);
+    std::int64_t o1, o2, o3;
+    o1 = (xf1 << 1) - 1;
+    o2 = (xf2 << 1) - 1;
+    o3 = (xf3 << 1) - 1;
+    LogicalLocation floc = nloc;
+    int lev = floc.level - root_level - 1;
+    floc.lx1 += o1;
+    bool fskip = false;
+    if (floc.lx1 < 0) {
+      if (mesh_bcs[BoundaryFace::inner_x1] == BoundaryFlag::periodic)
+        floc.lx1 = (nrbx1 << (lev+1)) - 1;
+      else
+        fskip = true;
+    } else if (floc.lx1 >= (nrbx1 << (lev+1))) {
+      if (mesh_bcs[BoundaryFace::outer_x1] == BoundaryFlag::periodic)
+        floc.lx1 = 0;
+      else
+        fskip = true;
+    }
+    if (!fskip) {
+      auto itr = locmap_[lev].find(floc);
+      if (itr != locmap_[lev].end()) {
+        const auto& it = *itr;
+        int i = it.second;
+        if (ranklist[i] == Globals::my_rank) {
+          ffc_send_.emplace_back(i, n, xf1^1, sffc1, -1);
+          if (newrank[n] == Globals::my_rank)
+            ffc_recv_.emplace_back(i, n, xf1, 0, s);
+          s++;
+        } else if (newrank[n] == Globals::my_rank) {
+          ffc_recv_.emplace_back(i, n, xf1, sffc1, -1);
+        }
+      }
+    }
+    if (f2) {
+      floc.lx1 = nloc.lx1, floc.lx2 += o2;
+      fskip = false;
+      if (floc.lx2 < 0) {
+        if (mesh_bcs[BoundaryFace::inner_x2] == BoundaryFlag::periodic)
+          floc.lx2 = (nrbx2 << (lev+1)) - 1;
+        else
+          fskip = true;
+      } else if (floc.lx2 >= (nrbx2 << (lev+1))) {
+        if (mesh_bcs[BoundaryFace::outer_x2] == BoundaryFlag::periodic)
+          floc.lx2 = 0;
+        else
+          fskip = true;
+      }
+      if (!fskip) {
+        auto itr = locmap_[lev].find(floc);
+        if (itr != locmap_[lev].end()) {
+          const auto& it = *itr;
+          int i = it.second;
+          if (ranklist[i] == Globals::my_rank) {
+            ffc_send_.emplace_back(i, n, xf2^1 + 2, sffc2, -1);
+            if (newrank[n] == Globals::my_rank)
+              ffc_recv_.emplace_back(i, n, xf2 + 2, 0, s);
+            s++;
+          } else if (newrank[n] == Globals::my_rank) {
+            ffc_recv_.emplace_back(i, n, xf2 + 2, sffc2, -1);
+          }
+        }
+      }
+    }
+    if (f3) {
+      floc.lx2 = nloc.lx2, floc.lx3 += o3;
+      fskip = false;
+      if (floc.lx3 < 0) {
+        if (mesh_bcs[BoundaryFace::inner_x3] == BoundaryFlag::periodic)
+          floc.lx3 = (nrbx3 << (lev+1)) - 1;
+        else
+          fskip = true;
+      } else if (floc.lx3 >= (nrbx3 << (lev+1))) {
+        if (mesh_bcs[BoundaryFace::outer_x3] == BoundaryFlag::periodic)
+          floc.lx3 = 0;
+        else
+          fskip = true;
+      }
+      if (!fskip) {
+        auto itr = locmap_[lev].find(floc);
+        if (itr != locmap_[lev].end()) {
+          const auto& it = *itr;
+          int i = it.second;
+          if (ranklist[i] == Globals::my_rank) {
+            ffc_send_.emplace_back(i, n, xf3^1 + 4, sffc3, -1);
```

Comment:
```suggestion
            ffc_send_.emplace_back(i, n, xf3^(1 + 4), sffc3, -1);
```

Reply:
Oh wait... I made the same mistake as Chat-GPT. It should be (xf3^1)+4. How could it be working...!?

Reply:
would be curious to hear if you come up with an explanation of how it worked for you locally before... were you running only a few tests?

Reply:
This error affects only ghost cells in the x2 and x3 surfaces, and I was checking only the x1 surface. Another lesson that covering all the test cases is important.